packages/react/src/ReactElementValidator.js

```diff
@@ -365,13 +369,20 @@ export function jsxWithValidation(
     }
   }
 
-  if (hasOwnProperty.call(props, 'key')) {
-    if (__DEV__) {
-      console.error(
-        'React.jsx: Spreading a key to JSX is a deprecated pattern. ' +
-          'Explicitly pass a key after spreading props in your JSX call. ' +
-          'E.g. <ComponentName {...props} key={key} />',
-      );
+  if (__DEV__) {
+    if (warnAboutSpreadingKeyToJSX) {
+      if (hasOwnProperty.call(props, 'key')) {
+        const name = getComponentName(type) || 'ComponentName';
+        if (didWarnAboutSpreadingKey[name] !== true) {
+          didWarnAboutSpreadingKey[name] = true;
+          console.error(
+            'React.jsx: Spreading a key to JSX is a deprecated pattern. ' +
+              'Explicitly pass a key after spreading props in your JSX call. ' +
+              'E.g. <%s {...props} key={key} />',
+            name,
+          );
+        }
+      }
```

Comment:
What I don't like about this implementation, is if a common component is used 'wrongly' in several places, it'll warn only for the 'first' one it encounters. A dev will fix it, run code again, and then see the next one. SO it'll become a game of whack-a-mole to fix all the usages. The cache should probably be based on component+location.

Reply:
punting on the deduplication for now.